Drush 13 fails with: AssertionError: Instead of using replacing Drush's logger, use $this->add() on DrushLoggerManager to add a custom logger

**Describe the bug**
I'm here because I am seeing this exception after trying to update to Drush 13. 

The stack trace mentions no custom code. Everything emits from within vendor/drush. 

In addition, after seeing the exception, I removed all my custom Drush commands from my application, cleared cache via the admin UI, and the error still appears. 

The merge request mentioned in the Exception has been in Drush since v [11.0.1](https://github.com/drush-ops/drush/releases/tag/11.0.1). I downgraded to Drush 13.2, 13.1, and 13.0 and all of them have this issue. I cleared cache via admin UI since I can't run cache:rebuild, between each version change. 

When I first saw the exception, I looked at my custom Command Alterer which [followed the Docs](https://github.com/drush-ops/drush/blob/13.x/sut/modules/unish/woot/src/Drush/CommandInfoAlterers/WootCommandInfoAlterer.php) and did this:

```php
  use AutowireTrait;

  /**
   * The drush logger channel.
   */
  protected LoggerChannelInterface $logger;

  public function __construct(LoggerChannelFactoryInterface $loggerFactory) {
    $this->logger = $loggerFactory->get('drush');
  }
```

But as I mentioned, in troubleshooting the error, I have removed the command alterer and all other custom commands, so that is not the cause of the issue. There are no Drush commands in `custom/` and the stack trace is all local to vendor/drush.

<details><summary>Stack Trace</summary>
<p>

AssertionError: Instead of using replacing Drush's logger, use $this->add() on DrushLoggerManager to add a custom logger. See https://github.com/drush-ops/drush/pull/5022 in /var/www/html/vendor/drush/drush/src/Commands/DrushCommands.php on line 76 #0 /var/www/html/vendor/drush/drush/src/Commands/DrushCommands.php(76): assert()
#1 /var/www/html/vendor/drush/drush/src/Runtime/ServiceManager.php(480): Drush\Commands\DrushCommands->logger()
#2 /var/www/html/vendor/drush/drush/src/Boot/DrupalBoot8.php(282): Drush\Runtime\ServiceManager->inflect()
#3 /var/www/html/vendor/drush/drush/src/Boot/DrupalBoot8.php(218): Drush\Boot\DrupalBoot8->addDrupalModuleDrushCommands()
#4 /var/www/html/vendor/drush/drush/src/Boot/BootstrapManager.php(211): Drush\Boot\DrupalBoot8->bootstrapDrupalFull()
#5 /var/www/html/vendor/drush/drush/src/Boot/BootstrapManager.php(397): Drush\Boot\BootstrapManager->doBootstrap()
#6 /var/www/html/vendor/drush/drush/src/Application.php(219): Drush\Boot\BootstrapManager->bootstrapMax()
#7 /var/www/html/vendor/drush/drush/src/Application.php(185): Drush\Application->bootstrapAndFind()
#8 /var/www/html/vendor/symfony/console/Application.php(266): Drush\Application->find()
#9 /var/www/html/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun()
#10 /var/www/html/vendor/drush/drush/src/Runtime/Runtime.php(110): Symfony\Component\Console\Application->run()
#11 /var/www/html/vendor/drush/drush/src/Runtime/Runtime.php(40): Drush\Runtime\Runtime->doRun()
#12 /var/www/html/vendor/drush/drush/drush.php(140): Drush\Runtime\Runtime->run()
#13 /var/www/html/vendor/bin/drush.php(119): include('...')
#14 {main}

</p>
</details> 

**To Reproduce**
```
ddev composer require drush/drush:^13
```

**Expected behavior**
To be able to run basic Drush commands like `drush status` and `drush cache:rebuild` without error.

**Actual behavior**
Exception thrown.

**Workaround**
No. All Drush commands fail. 

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 13.3.3.0 (Also with 13.2, 13.1, 13.0) Bug does not appear with Drush 12.
| Drupal version? | 10.4.x
| PHP version     | 8.3
| OS?             | Mac/Linux/Windows (DDEV/linux)

**Additional information**
Add any other context about the problem here.

--- a/src/Runtime/DependencyInjection.php
+++ b/src/Runtime/DependencyInjection.php
@@ -5,6 +5,7 @@ declare(strict_types=1);
 namespace Drush\Runtime;
 
 use Composer\Autoload\ClassLoader;
+use Consolidation\AnnotatedCommand\CommandFileDiscovery;
 use Consolidation\Config\ConfigInterface;
 use Consolidation\Config\Util\ConfigOverlay;
 use Consolidation\SiteAlias\SiteAliasManager;
@@ -31,6 +32,7 @@ use League\Container\ContainerInterface;
 use Robo\Robo;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
 
 /**
  * Prepare our Dependency Injection Container
@@ -143,7 +145,7 @@ class DependencyInjection
         // but we will register and configure one for our use.
         // TODO: Some old adapter code uses this, but the Symfony dispatcher does not.
         // See Application::commandDiscovery().
-        Robo::addShared($container, 'commandDiscovery', 'Consolidation\AnnotatedCommand\CommandFileDiscovery')
+        Robo::addShared($container, 'commandDiscovery', CommandFileDiscovery::class)
             ->addMethodCall('addSearchLocation', ['CommandFiles'])
             ->addMethodCall('setSearchPattern', ['#.*(Commands|CommandFile).php$#']);
 
@@ -161,7 +163,7 @@ class DependencyInjection
     protected function alterServicesForDrush($container, Application $application): void
     {
         $paramInjection = $container->get('parameterInjection');
-        $paramInjection->register('Symfony\Component\Console\Style\SymfonyStyle', new DrushStyleInjector());
+        $paramInjection->register(SymfonyStyle::class, new DrushStyleInjector());
 
         // Add our own callback to the hook manager
         $hookManager = $container->get('hookManager');

--- a/src/Runtime/DependencyInjection.php
+++ b/src/Runtime/DependencyInjection.php
@@ -12,7 +12,9 @@ use Consolidation\SiteAlias\SiteAliasManagerAwareInterface;
 use Consolidation\SiteAlias\SiteAliasManagerInterface;
 use Consolidation\SiteProcess\ProcessManagerAwareInterface;
 use Drush\Application;
+use Drush\Boot\BootstrapHook;
 use Drush\Boot\BootstrapManager;
+use Drush\Boot\DrupalBoot8;
 use Drush\Cache\CommandCache;
 use Drush\Command\DrushCommandInfoAlterer;
 use Drush\Command\GlobalOptionsEventListener;
@@ -116,25 +118,25 @@ class DependencyInjection
             ->addMethodCall('addSimplifier', [new EntityToArraySimplifier()]);
 
         // Add some of our own objects to the container
-        Robo::addShared($container, 'service.manager', 'Drush\Runtime\ServiceManager')
+        Robo::addShared($container, 'service.manager', ServiceManager::class)
             ->addArgument(self::LOADER)
             ->addArgument('config')
             ->addArgument('logger');
-        Robo::addShared($container, 'bootstrap.drupal8', 'Drush\Boot\DrupalBoot8')
+        Robo::addShared($container, 'bootstrap.drupal8', DrupalBoot8::class)
             ->addArgument('service.manager')
             ->addArgument(self::LOADER);
-        Robo::addShared($container, self::BOOTSTRAP_MANAGER, 'Drush\Boot\BootstrapManager')
+        Robo::addShared($container, self::BOOTSTRAP_MANAGER, BootstrapManager::class)
             ->addMethodCall('setDrupalFinder', [$drupalFinder])
             ->addMethodCall('add', ['bootstrap.drupal8']);
         Robo::addShared($container, BootstrapManager::class, self::BOOTSTRAP_MANAGER); // For autowiring
-        Robo::addShared($container, 'bootstrap.hook', 'Drush\Boot\BootstrapHook')
+        Robo::addShared($container, 'bootstrap.hook', BootstrapHook::class)
           ->addArgument(self::BOOTSTRAP_MANAGER);
-        Robo::addShared($container, 'tildeExpansion.hook', 'Drush\Runtime\TildeExpansionHook');
+        Robo::addShared($container, 'tildeExpansion.hook', TildeExpansionHook::class);
         Robo::addShared($container, 'process.manager', ProcessManager::class)
             ->addMethodCall('setConfig', ['config'])
             ->addMethodCall('setConfigRuntime', ['config.runtime'])
             ->addMethodCall('setDrupalFinder', [$drupalFinder]);
-        Robo::addShared($container, 'redispatch.hook', 'Drush\Runtime\RedispatchHook')
+        Robo::addShared($container, 'redispatch.hook', RedispatchHook::class)
             ->addArgument('process.manager');
 
         // Robo does not manage the command discovery object in the container,
@@ -146,8 +148,8 @@ class DependencyInjection
             ->addMethodCall('setSearchPattern', ['#.*(Commands|CommandFile).php$#']);
 
         // Error and Shutdown handlers
-        Robo::addShared($container, 'errorHandler', 'Drush\Runtime\ErrorHandler');
-        Robo::addShared($container, 'shutdownHandler', 'Drush\Runtime\ShutdownHandler');
+        Robo::addShared($container, 'errorHandler', ErrorHandler::class);
+        Robo::addShared($container, 'shutdownHandler', ShutdownHandler::class);
 
         // Add inflectors. @see \Drush\Boot\BaseBoot::inflect
         $container->inflector(SiteAliasManagerAwareInterface::class)

--- a/sut/modules/unish/woot/src/Drush/CommandInfoAlterers/WootCommandInfoAlterer.php
+++ b/sut/modules/unish/woot/src/Drush/CommandInfoAlterers/WootCommandInfoAlterer.php
@@ -6,20 +6,12 @@ namespace Drupal\woot\Drush\CommandInfoAlterers;
 
 use Consolidation\AnnotatedCommand\CommandInfoAltererInterface;
 use Consolidation\AnnotatedCommand\Parser\CommandInfo;
-use Drupal\Core\Logger\LoggerChannelFactoryInterface;
-use Drupal\Core\Logger\LoggerChannelInterface;
-use Drush\Commands\AutowireTrait;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
 
-class WootCommandInfoAlterer implements CommandInfoAltererInterface
+class WootCommandInfoAlterer implements CommandInfoAltererInterface, LoggerAwareInterface
 {
-    use AutowireTrait;
-
-    protected LoggerChannelInterface $logger;
-
-    public function __construct(LoggerChannelFactoryInterface $loggerFactory)
-    {
-        $this->logger = $loggerFactory->get('drush');
-    }
+    use LoggerAwareTrait;
 
     public function alterCommandInfo(CommandInfo $commandInfo, $commandFileInstance)
     {

